Use black to check for proper code formatting and reformat existing code

barthalion opened the pull request.

barthalion committed:
Use black to check for proper code formatting

wjt commented on src/checkers/firefoxchecker.py:
```diff
@@ -39,44 +39,50 @@
-    CHECKER_BROWSER_DEFAULT_LANGUAGE = 'en-US'
-    CHECKER_TRANSLATIONS_FILENAME_MATCH = r'.+\.xpi$'
+    FIREFOX_RELEASE_INFO_URL = (
+        "https://product-details.mozilla.org/1.0/firefox_versions.json"  # noqa: E501
```
```suggestion
        "https://product-details.mozilla.org/1.0/firefox_versions.json"
```

at which point it probably fits on one line again

barthalion replied:
Will look into that tomorrow.

barthalion replied:
It doesn't seem to affect the way Black formats it.

wjt commented on src/checkers/firefoxchecker.py:
```diff
@@ -39,44 +39,50 @@
+        "https://product-details.mozilla.org/1.0/firefox_versions.json"  # noqa: E501
+    )
+    FIREFOX_ARCHIVE_BASE_URL = (
+        "https://archive.mozilla.org/pub/firefox/releases/{version}/"  # noqa: E501
```
```suggestion
        "https://archive.mozilla.org/pub/firefox/releases/{version}/"
```

wjt commented on src/checkers/firefoxchecker.py:
```diff
@@ -39,44 +39,50 @@
-                        module_data.name,
-                        self.CHECKER_BROWSER_SOURCE_FILENAME)
+            log.warning(
+                "Unable to find browser source with filename %s",
```
The reformatting here is correct but it has shown that the log message is wrong – it only has one `%s` placeholder.

barthalion replied:
I might be wrong but original code seems to have only one placeholder as well.

wjt replied:
Yes, I know. Up to you if you fancy fixing it.

barthalion replied:
Ah, sure, misunderstood your comment.

wjt commented on src/main.py:
```diff
@@ -174,34 +176,39 @@ def open_pr(subject, body, branch):
+    parser.add_argument(
+        "--update",
+        help="Update manifest(s) to refer to new versions of "
+        "external data - also open PRs for changes unless "
```
I really, really loathe this aspect of `black`'s formatting, but still not enough to do anything about it like “file a bug”.

barthalion replied:
Yeah, I'm not particularly happy about it either, but it hand-waves the entire class of code formatting issues like `gofmt` does. I can live with opinionated tool that makes me not think about it even if I dislike some things.

wjt reviewed (changes requested):
OK! I've been doing this piecemeal (as you may be able to guess from `.flake8`) but I guess we should bite this bullet at some point. That said:

barthalion committed:
Reformat code with black

barthalion committed:
firefoxchecker: Remove unused argument passed to log

wjt committed:
tests: fix Xdebug scraper test

[As I predicted][0] when this test was added, this test started failing
because 2.9.0 is no longer the latest version of Xdebug and so is not on
the front page any more.

[0]: https://github.com/flathub/flatpak-external-data-checker/pull/46#discussion_r359545199
